qiskit_experiments/framework/base_experiment.py

```diff
@@ -512,46 +511,17 @@ def set_analysis_options(self, **fields):
     def _metadata(self) -> Dict[str, any]:
         """Return experiment metadata for ExperimentData.
 
-        The :meth:`_add_job_metadata` method will be called for each
-        experiment execution to append job metadata, including current
-        option values, to the ``job_metadata`` list.
+        Subclasses can override this method to add custom experiment
+        metadata to the returned experiment result data.
         """
-        metadata = {
-            "experiment_type": self._type,
-            "num_qubits": self.num_qubits,
-            "physical_qubits": list(self.physical_qubits),
-        }
-        # Add additional metadata if subclasses specify it
-        for key, val in self._additional_metadata().items():
-            metadata[key] = val
-        return metadata
-
-    def _additional_metadata(self) -> Dict[str, any]:
-        """Add additional subclass experiment metadata.
-
-        Subclasses can override this method if it is necessary to store
-        additional experiment metadata in ExperimentData.
-        """
-        return {}
+        metadata = {"physical_qubits": list(self.physical_qubits)}
 
-    def _add_job_metadata(self, metadata: Dict[str, Any], jobs: BaseJob, **run_options):
-        """Add runtime job metadata to ExperimentData.
-
-        Args:
-            metadata: the metadata dict to update with job data.
-            jobs: the job objects.
-            run_options: backend run options for the job.
-        """
-        values = {
-            "job_ids": [job.job_id() for job in jobs],
-            "experiment_options": copy.copy(self.experiment_options.__dict__),
-            "transpile_options": copy.copy(self.transpile_options.__dict__),
-            "run_options": copy.copy(run_options),
-        }
-        if self.analysis is not None:
-            values["analysis_options"] = copy.copy(self.analysis.options.__dict__)
-
-        metadata["job_metadata"] = [values]
+        # Store measurement level and meas return if they have been
+        # set for the experiment
+        for run_opt in ["meas_level", "meas_return"]:
+            if hasattr(self.run_options, run_opt):
+                metadata[run_opt] = getattr(self.run_options, run_opt)
```

Comment:
```suggestion
                metadata[run_opt] = self.run_options.get(run_opt)
```
they have get method.